### PR TITLE
[#4695] Fix (python-client): Fix gradlew :clients:client-python:build failed if the package does not contain .git directory

### PR DESCRIPTION
**Pull request by diqiu50**
### What changes were proposed in this pull request?

Fix  gradlew :clients:client-python:build failed if the package does not contain .git directory

### Why are the changes needed?

Fix: #4695 

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

Manually test

### CODE REVIEW

**Comment by yuqi1129** [comment]
LGTM, let wait the CI pass.

File: clients/client-python/scripts/generate_version.py [hunk 1]
```diff
@@ -17,17 +17,36 @@
...
+        with open(git_path + "HEAD", "r", encoding="utf-8") as file:
+            ref = file.readline().strip()
+
+        if ref.startswith("ref:"):
+            ref_path = ref.split(" ")[1]
+            with open(git_path + ref_path, "r", encoding="utf-8") as file:
+                commit_id = file.readline().strip()
+        return commit_id
+    except (FileNotFoundError, IOError) as e:
+        print(f"Warn: Failed to get git commit Id. {e}")
```
Comment (jerryshao) [inline clients/client-python/scripts/generate_version.py:46]: I think we should not use `print`, either we use log, or swallow the exception.

Comment (diqiu50) [reply clients/client-python/scripts/generate_version.py]: I remove it

**Comment by mchades** [comment]
plz fix the code style to make CI passed. @diqiu50

**Review by jerryshao — APPROVED** [review]
LGTM.

**Pull request merged**